File: lib-mvpbase-appcompat/src/main/java/com/baozi/mvp/templet/TemplateLoadingFragment.java
package com.baozi.mvp.templet;

import android.view.View;

import com.baozi.mvp.MVPManager;
import com.baozi.mvp.presenter.BasePresenter;
import com.baozi.mvp.templet.options.ContentOptions;
import com.baozi.mvp.templet.weight.LoadingPager;
import com.baozi.mvp.view.LoadView;

/**
 * Created by baozi on 2017/12/20.
 */

public abstract class TemplateLoadingFragment<T extends BasePresenter> extends TemplateFragment<T>
        implements LoadView {
    private LoadingPager mLoadingPager;

    @Override
    protected View wrapperContentView(View view) {
        mLoadingPager = getContentOptions().buildLoadingPager(mContext, view);
        mLoadingPager.setRefreshListener(new LoadingPager.OnRefreshListener() {
            @Override
            public void onRefresh() {
                triggerInit();
            }
        });
        return mLoadingPager;
    }

    @Override
    public void showEmpty() {
        mLoadingPager.showEmpty();
    }

    @Override
    public void showError(Throwable throwable, boolean isShowError) {
        if (isShowError) {
            mLoadingPager.showError(throwable);
        }
    }

    @Override
    public void showError(Throwable throwable) {
        showError(throwable, true);
    }

    @Override
    public void onNewThrowable(Throwable throwable) {
        super.onNewThrowable(throwable);
        showError(throwable);
    }


    @Override
    public void showLoading() {
        mLoadingPager.showLoading();
    }

    @Override
    public void showSuccess() {
        mLoadingPager.showSuccess();
    }

    @Override
    public void triggerInit() {
        showLoading();
        mPresenter.onRefreshData();
    }

    @Override
    public LoadingPager getLoadPager() {
        return mLoadingPager;
    }

}

File: lib-mvpbase-kotlin/src/main/java/com/linfeng/mvp/templet/TemplateLoadingFragment.kt
//package com.linfeng.mvp.templet
//
//import android.view.View
//import com.linfeng.mvp.MVPManager
//import com.linfeng.mvp.presenter.BasePresenter
//import com.linfeng.mvp.templet.options.ContentOptions
//import com.linfeng.mvp.templet.weight.LoadingPager
//import com.linfeng.mvp.view.LoadView
//
//
///**
// * Created by baozi on 2017/12/20.
// */
//
//abstract class TemplateLoadingFragment<T : BasePresenter<*>> : TemplateFragment<T>(), LoadView {
//    var loadPager: LoadingPager? = null
//        private set
//
//    protected override val contentOptions: ContentOptions
//        get() = MVPManager.getContentOptions()
//
//    override fun wrapperContentView(view: View): View {
//        loadPager = contentOptions.buildLoadingPager(mContext, view)
//        loadPager!!.setRefreshListener(object : LoadingPager.OnRefreshListener() {
//            fun onRefresh() {
//                triggerInit()
//            }
//        })
//        return loadPager
//    }
//
//    fun showEmpty() {
//        loadPager!!.showEmpty()
//    }
//
//    @JvmOverloads
//    fun showError(throwable: Throwable, isShowError: Boolean = true) {
//        if (isShowError) {
//            loadPager!!.showError(throwable)
//        }
//        onNewThrowable(throwable)
//    }
//
//    fun showLoading() {
//        loadPager!!.showLoading()
//    }
//
//    fun showSuccess() {
//        loadPager!!.showSuccess()
//    }
//
//    fun triggerInit() {
//        mPresenter.onRefreshData()
//    }
//}
